### PR TITLE
Add Vsphere snapshotter

### LINKED ISSUE #9013: Implement CSI Snapshotter
### What would you like to be added? / User Story
CSI Snapshotter is part of the CSI Spec and CSI driver may require it. See: https://github.com/kubermatic/kubermatic/issues/8928

https://github.com/kubernetes-csi/external-snapshotter

The snapshotter should be independent of the csi driver, which does not always seem to be the case.

It consist of 3 components: 

- Snapshot CRD
- Snapshot Controller
- Validating Webhook

### Acceptance criteria
* Make sure all mentioned components are deployed together with the csi driver
* Note that some csi driver provide some of the components as part of their csi driver

### PR DESCRIPTION
**Pull request by mate4st**
**What does this PR do / Why do we need it**:

Adds the snapshot controller and validation webhook for vsphere csi

Fixes #9013

**Documentation**:
https://github.com/kubernetes-csi/external-snapshotter
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/manifests/vanilla/deploy-csi-snapshot-components.sh

```release-note
Add vSphere Snapshotter
```

### CODE REVIEW

File: pkg/controller/user-cluster-controller-manager/resources/resources/csi-snapshotter/secret.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package csi
+package csisnaphotter
```
Comment (xrstf) [inline pkg/controller/user-cluster-controller-manager/resources/resources/csi-snapshotter/secret.go]: Typo in package name

**Review by xrstf — CHANGES_REQUESTED** [review]
Typonista

File: pkg/controller/user-cluster-controller-manager/resources/resources/csi-snapshotter/secret.go (hunk 1 above)
Comment (mate4st) [reply pkg/controller/user-cluster-controller-manager/resources/resources/csi-snapshotter/secret.go]: Its obviously new component, the snap Hotter :fire:

**Comment by xrstf** [comment]
Those errors will be fixed once #9420 is merged. Sorry 'bout that.

**Pull request merged**